src/main/java/gil/server/Response.java

```diff
@@ -32,8 +30,53 @@ public String getContentLength() {
         return contentLength;
     }
 
-    public String getBody() {
-        return body;
+    public String getDate() {
+        return date;
+    }
+
+    public String getProtocol() {
+        return protocol;
     }
 
+    public String getReasonPhrase() {
+        return reasonPhrase;
+    }
+
+    public String getStartLine() {
+        return protocol + " " + statusCode + " " + reasonPhrase;
+    }
+
+    public String getStatusCode() {
+        return statusCode;
+    }
+
+    public void setBody(String body) throws UnsupportedEncodingException {
+        this.body = body;
+        setContentLength(this.body);
+    }
+
+    private void setContentLength (String body) throws UnsupportedEncodingException {
+        byte[] responseBytes = body.getBytes("UTF-8");
+        this.contentLength = "Content-Length: " + responseBytes.length;
+    }
+
+    public void setContentType (String body) {
+        this.contentType = body;
+    }
+
+    private void setDate() {
+        this.date = "Date: " + DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now(ZoneOffset.UTC));
```

Comment:
Instead of chaining this together, I'd recommend breaking it up for easier readability purposes:
```
ZonedDateTime currentUTCDateTime = ZonedDateTime.now(ZoneOffset.UTC);
String RFC1123FormattedDate = DateTimeFormatter.RFC_1123_DATE_TIME.format(currentUTCDateTime);
this.date = "Date: " + RFC1123FormattedDate;
```